File: Lib/Telerik/Examples/DemoAppsHub/GridExample/GridExampleUserControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports Telerik.Charting
Imports Telerik.WinControls.UI
Imports Telerik.WinControls.Themes

Namespace DemoAppsHub.GridExample
	Public Partial Class GridExampleUserControl
		Inherits UserControl
		Public Sub New()
			InitializeComponent()
		End Sub

		#Region "Event handlers"

		Protected Overrides Sub OnLoad(ByVal e As EventArgs)
			MyBase.OnLoad(e)

			Me.radGridView1.ThemeName = "TelerikMetroBlue"
			AddHandler radGridView1.CreateRow, AddressOf radGridView1_CreateRow
			AddHandler radGridView1.MasterTemplate.CreateRowInfo, AddressOf MasterTemplate_CreateRowInfo

			Me.employeesTableAdapter.Fill(Me.northwindDataSet.Employees)
			Me.radGridView1.TableElement.RowHeight = 45
		End Sub

		Private Sub MasterTemplate_CreateRowInfo(ByVal sender As Object, ByVal e As GridViewCreateRowInfoEventArgs)
			e.RowInfo.MinHeight = 45
		End Sub

		Private Sub radGridView1_CreateRow(ByVal sender As Object, ByVal e As GridViewCreateRowEventArgs)
			If e.RowType Is GetType(GridDataRowElement) Then
				e.RowType = GetType(GridCustomRowElement)
			End If
		End Sub

		#End Region
	End Class
End Namespace

File: Lib/Telerik/Examples/DemoAppsHub/GridExample/GridExampleUserControl.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using Telerik.Charting;
using Telerik.Charting.Styles;
using Telerik.WinControls.UI;
using Telerik.WinControls.Themes;

namespace DemoAppsHub.GridExample
{
    public partial class GridExampleUserControl : UserControl
    {
        public GridExampleUserControl()
        {
            InitializeComponent();            
        }

        #region Event handlers

        protected override void OnLoad(EventArgs e)
        {
            base.OnLoad(e);

            this.radGridView1.ThemeName = "TelerikMetroBlue";
            this.radGridView1.CreateRow += new GridViewCreateRowEventHandler(radGridView1_CreateRow);
            this.radGridView1.MasterTemplate.CreateRowInfo += new GridViewCreateRowInfoEventHandler(MasterTemplate_CreateRowInfo);

            this.employeesTableAdapter.Fill(this.northwindDataSet.Employees);
            this.radGridView1.TableElement.RowHeight = 45;
        }

        void MasterTemplate_CreateRowInfo(object sender, GridViewCreateRowInfoEventArgs e)
        {
            e.RowInfo.MinHeight = 45;
        }

        void radGridView1_CreateRow(object sender, GridViewCreateRowEventArgs e)
        {
            if (e.RowType == typeof(GridDataRowElement))
            {
                e.RowType = typeof(GridCustomRowElement);
            }
        }

        #endregion
    }
}
